domino/datasets.py

```diff
@@ -0,0 +1,190 @@
+import hashlib
+import math
+import os
+import time
+from concurrent.futures import ThreadPoolExecutor
+from logging import Logger
+from typing import AnyStr
+
+from domino.http_request_manager import _HttpRequestManager
+from domino.routes import _Routes
+
+FILE_UPLOAD_SETTING_DEFAULT = "Overwrite"
+MAX_WORKERS = 10
+MAX_UPLOAD_ATTEMPTS = 10
+MB = 2 ** 20  # 2^20 bytes - 1 Megabyte
+SLEEP_TIME_IN_SEC = 5
+UPLOAD_READ_TIMEOUT_IN_SEC = 30
+
+
+class UploadChunk:
+    def __init__(
+            self,
+            absolute_path: str,
+            chunk_number: int,
+            dataset_id: str,
+            file_name: str,
+            file_size: int,
+            upload_key: str,
+            identifier: str,
+            relative_path: str,
+            target_chunk_size: int,
+            total_chunks: int,
+    ):
+        self.absolute_path = absolute_path
+        self.chunk_number = chunk_number
+        self.dataset_id = dataset_id
+        self.file_name = file_name
+        self.file_size = file_size
+        self.identifier = identifier
+        self.relative_path = relative_path
+        self.target_chunk_size = target_chunk_size
+        self.total_chunks = total_chunks
+        self.upload_key = upload_key
```

Comment:
Small Nit: Might be clearer to match the order or the arguments of `__init__` with the order the variables are assigned.